packages/esm-patient-registration-app/src/patient-registration/patient-registration.test.tsx

```diff
@@ -173,264 +174,272 @@ const fillRequiredFields = async () => {
-    const weight = within(customSection).getByLabelText('Weight (kg)');
-    await user.type(weight, '-999');
+      await user.click(screen.getByText('Register Patient'));
 
-    mockedSaveEncounter.mockRejectedValue({ status: 400, responseBody: { error: { message: 'an error message' } } });
+      await waitFor(() => expect(mockedSavePatient).toHaveBeenCalledTimes(1));
+      await waitFor(() => expect(mockedSaveEncounter).toHaveBeenCalledTimes(1));
+      await waitFor(() =>
+        expect(mockedShowToast).toHaveBeenCalledWith(expect.objectContaining({ description: 'an error message' })),
+      );
+
+      mockedSaveEncounter.mockResolvedValue({});
+
+      await user.click(screen.getByText('Register Patient'));
+      await waitFor(() => expect(mockedSavePatient).toHaveBeenCalledTimes(2));
+      await waitFor(() => expect(mockedSaveEncounter).toHaveBeenCalledTimes(2));
+      await waitFor(() => expect(mockedShowToast).toHaveBeenCalledWith(expect.objectContaining({ kind: 'success' })));
+    });
+  });
 
-    await user.click(screen.getByText('Register Patient'));
+  describe('when updating an existing patient details', () => {
+    beforeEach(() => {
+      mockedUseConfig.mockReturnValue(mockOpenmrsConfig);
+      mockedSavePatient.mockReturnValue({ data: { uuid: 'new-pt-uuid' }, ok: true });
+      mockedSaveEncounter.mockClear();
+      mockedShowToast.mockClear();
+      jest.clearAllMocks();
+    });
 
-    await waitFor(() => expect(mockedSavePatient).toHaveBeenCalledTimes(1));
-    await waitFor(() => expect(mockedSaveEncounter).toHaveBeenCalledTimes(1));
-    await waitFor(() =>
-      expect(mockedShowToast).toHaveBeenCalledWith(expect.objectContaining({ description: 'an error message' })),
-    );
+    it('edits patient demographics', async () => {
+      const user = userEvent.setup();
 
-    mockedSaveEncounter.mockResolvedValue({});
+      mockedSavePatient.mockResolvedValue({});
 
-    await user.click(screen.getByText('Register Patient'));
-    await waitFor(() => expect(mockedSavePatient).toHaveBeenCalledTimes(1));
-    await waitFor(() => expect(mockedSaveEncounter).toHaveBeenCalledTimes(2));
-    await waitFor(() => expect(mockedShowToast).toHaveBeenCalledWith(expect.objectContaining({ kind: 'success' })));
+      const mockedUseParams = useParams as jest.Mock;
+
+      mockedUseParams.mockReturnValue({ patientUuid: mockPatient.id });
+
+      mockedUsePatient.mockReturnValue({
+        isLoading: false,
+        patient: mockPatient,
+        patientUuid: mockPatient.id,
+        error: null,
+      });
+
+      render(
+        <ResourcesContext.Provider value={mockResourcesContextValue}>
+          <Router>
+            <PatientRegistration isOffline={false} savePatientForm={mockedSavePatient} />
+          </Router>
+        </ResourcesContext.Provider>,
+      );
+
+      const givenNameInput = screen.getByLabelText(/First Name/) as HTMLInputElement;
```

Comment:
This might also work:
```suggestion
      const givenNameInput: HTMLInputElement = screen.findByLabelText(/First Name/);
```